stonesoup/updater/probability.py

```diff
@@ -0,0 +1,192 @@
+                posterior_states.append(hypothesis.prediction)
+            else:
+                posterior_state = super().update(hypothesis, **kwargs)  # Use the EKF update
+                posterior_states.append(posterior_state)
+            posterior_state_weights.append(hypothesis.probability)
+
+        means = StateVectors([state.state_vector for state in posterior_states])
+        covars = np.stack([state.covar for state in posterior_states], axis=2)
+        weights = np.asarray(posterior_state_weights)
+
+        # Reduce mixture of states to one posterior estimate Gaussian.
+        post_mean, post_covar = gm_reduce_single(means, covars, weights)
+
+        return post_mean, post_covar
+
+    def _update_via_innovation(self, hypotheses, **kwargs):
+        """Of n hypotheses there should be 1 prediction (a missed detection hypothesis) and n-1
+        different measurement associations. The update proceeds as described above.
+
+        Parameters
+        ----------
+        hypotheses : :class:`~.MultipleHypothesis`
+            As in :meth:`update` method
+         **kwargs : various
+            These are passed to :meth:`predict_measurement`
+
+        Returns
+        -------
+        : :class:`~.StateVector`
+            The mean of the reduced/single Gaussian
+        : :class:`~.CovarianceMatrix`
+            The covariance of the reduced/single Gaussian
+        """
+
+        for n, hypothesis in enumerate(hypotheses):
+            # Check for the existence of an associated measurement. Because of the way the
+            # hypothesis is constructed, you can do this:
+            if not hypothesis:
+                hypothesis.measurement_prediction = self.predict_measurement(
+                    hypothesis.prediction, **kwargs)
+                innovation = hypothesis.measurement_prediction.state_vector - \
+                    hypothesis.measurement_prediction.state_vector  # is zero in this case
+                posterior_covariance, kalman_gain = self._posterior_covariance(hypothesis)
+                # Add the weighted prediction to the weighted posterior
+                posterior_covariance = float(hypothesis.probability) * \
+                    hypothesis.prediction.covar + (1 - float(hypothesis.probability)) * \
+                    posterior_covariance
+                posterior_mean = copy(hypothesis.prediction.state_vector)
+            else:
+                innovation = hypothesis.measurement.state_vector - \
+                             hypothesis.measurement_prediction.state_vector
+
+            # probably exists a less clunky way of doing this using exists() or overwritten +=
+            # All these floats should be redundant if/when the bug in Probability.__mult__() is
+            # fixed.
+            if n == 0:
+                sum_of_innovations = float(hypothesis.probability) * innovation
+                sum_of_weighted_cov = float(hypothesis.probability) * (innovation @ innovation.T)
```

Comment:
Set these two sum variables outside the loop to `0`?

Reply:
Wasn't too sure if the proposed would work, but doing a simple test, looks like it should.
```
>>> a = 0
>>> a += np.array([[0], [0]])
>>> a
array([[0],
       [0]])
```

Reply:
So, 
`sum_of_innovations = 0
sum_of_innovations += hypothesis.probability * innovation`
?

Reply:
> So, `sum_of_innovations = 0` `sum_of_innovations += hypothesis.probability * innovation` ?

But won't matter if :class:`Probability` is corrected.

Reply:
```diff
diff --git a/stonesoup/updater/probability.py b/stonesoup/updater/probability.py
index c471e514..c6495111 100644
--- a/stonesoup/updater/probability.py
+++ b/stonesoup/updater/probability.py
@@ -156,6 +156,8 @@ class PDAUpdater(ExtendedKalmanUpdater):
             The covariance of the reduced/single Gaussian
         """
 
+        sum_of_innovations = 0
+        sum_of_weighted_cov = 0
         for n, hypothesis in enumerate(hypotheses):
             # Check for the existence of an associated measurement. Because of the way the
             # hypothesis is constructed, you can do this:
@@ -174,15 +176,10 @@ class PDAUpdater(ExtendedKalmanUpdater):
                 innovation = hypothesis.measurement.state_vector - \
                              hypothesis.measurement_prediction.state_vector
 
-            # probably exists a less clunky way of doing this using exists() or overwritten +=
             # All these floats should be redundant if/when the bug in Probability.__mult__() is
             # fixed.
-            if n == 0:
-                sum_of_innovations = float(hypothesis.probability) * innovation
-                sum_of_weighted_cov = float(hypothesis.probability) * (innovation @ innovation.T)
-            else:
-                sum_of_innovations += float(hypothesis.probability) * innovation
-                sum_of_weighted_cov += float(hypothesis.probability) * (innovation @ innovation.T)
+            sum_of_innovations += float(hypothesis.probability) * innovation
+            sum_of_weighted_cov += float(hypothesis.probability) * (innovation @ innovation.T)
 
         posterior_mean += kalman_gain @ sum_of_innovations
         posterior_covariance += \
```